pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -217,7 +257,8 @@ func (c *Cluster) completeOSDsForAllNodes(config *provisionConfig, configOSDs bo
 						logger.Infof("skipping event from node %s status update since it is already completed", node)
 						continue
 					}
-					completed := c.handleStatusConfigMapStatus(node, config, configMap, configOSDs)
+					var completed bool
+					completed, deferredConfigMaps = c.handleStatusConfigMapStatus(node, config, configMap, deferredConfigMaps, configOSDs)
 					if completed {
 						remainingNodes.Remove(node)
 						if remainingNodes.Count() == 0 {
```

Comment:
Will the remaining nodes count be accurate since some are deferred? The next line is important (at least for me) to see in the logs when osds are done provisioning. Seems like we can't print this until the deferred osds are also updated.

Reply:
I think this confusion is indicative of how complicated the OSD orchestration loop is. But the short is that nodes (actually PVCs) that don't get provision jobs started for them are already marked as completed and don't get added to the `remainingNodes` list. This happens in the `handleStatusConfigMapStatus` fxn.

I'm hoping that when I go and implement the OSD updates by failure domain, there will be good opportunity to clear up some of the inadvertent obfuscation we have in the code here.